sql/core/src/main/scala/org/apache/spark/sql/execution/joins/ShuffledHashJoinExec.scala

```diff
@@ -484,7 +496,10 @@ case class ShuffledHashJoinExec(
          |    }
          |  }
          |
-         |  $consumeFullOuterJoinRow();
+         |  if ($foundMatch || $isFullOuterJoin) {
```

Comment:
nit: we can statically remove or simply the condition:
```
val consumeCode = if (isFullOuterJoin) {
  s"$consumeOuterJoinRow();"
} else {
  s"""
    |if ($foundMatch) {
    |  $consumeOuterJoinRow();
    |}
  """
}
```

Reply:
Thanks , let me make some follow ups a bit later